### PR TITLE
fix: dns resolution in fedora

### PR DESCRIPTION
**Pull request by gouravkrosx**
## Related Issue
  - Sync some commit & sync commit (fix: DNS resolution in fedora)

Closes: #[issue number that will be closed through this PR]

#### Describe the changes you've made
- Sync commit 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Please let us know if any test cases are added

Please describe the tests(if any). Provide instructions how its affecting the coverage.

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
A clear and concise description of it.

## Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## Screenshots (if any)

|        Original         |          Updated           |
|:-----------------------:|:--------------------------:|
| **original screenshot** | <b>updated screenshot </b> |

### CODE REVIEW

File: pkg/core/core.go [hunk 1]
```diff
@@ -117,6 +120,18 @@ func (c *Core) Hook(ctx context.Context, id uint64, _ models.HookOptions) error
 			utils.LogError(c.logger, err, "failed to unload the hooks")
 		}
 
+		if opts.Mode != models.MODE_TEST {
```
Comment (charankamarapu) [inline pkg/core/core.go:123]: you can use 
```
if opts.Mode == models.MODE_TEST  &&  c.hostConfigStr != "" {
}
```
current thing is a bit hard to read

File: pkg/core/core.go [hunk 2]
```diff
@@ -157,6 +172,16 @@ func (c *Core) Hook(ctx context.Context, id uint64, _ models.HookOptions) error
 	}
 
 	c.proxyStarted = true
+
+	if opts.Mode != models.MODE_TEST {
```
Comment (charankamarapu) [inline pkg/core/core.go:176]: same here

**Review by charankamarapu — COMMENTED** [review]
everything LGTM, Please resolve the comment.

File: pkg/core/core.go (hunk 1 above)
Comment (gouravkrosx) [reply pkg/core/core.go:123]: cool

**Review by charankamarapu — APPROVED** [review]
ship it!

**Pull request merged**